XU-TechnicalPaper.md

```diff
@@ -154,19 +158,85 @@ Market makers are required to stake a certain amount of XUC in a smart contract
- 
-It is evident, that the approach to enforce direct socket connections between peers to exchange order book information can't scale infinitely. We are targeting to establish a relay network of 'super nodes' in a later phase, where entities running an XUD can choose to forward and receive order book information from these super nodes. To become a super node for a peer, it has to provide a faster connection between two specific peers as a base requirement. This can be achieved for example through a strategic location on a fiber connection, where peers wouldn't have direct access to and help to further scale and speed-up order book information exchange.
 
-Also, there will be no complete global order book as such, containing all orders for each and every trading pair. Instead, orders will be propagated based on an XUDs preferences, e.g. only specific trading pairs or only from specific peers. Therefore, XUDs only maintain relevant parts of the order book. This approach ensures a more efficient system. XUDs constantly exchange order information with connected peers.
+Also, there will be no complete global order book as such, containing all orders for each and every trading pair. Instead, orders will be propagated based on an XUDs preferences, e.g. only specific trading pairs or only from specific peers. Therefore, XUDs only maintain relevant parts of the order book. XUDs constantly exchange order information and updates with connected peers.
+
+[XUD Order Book Data Structure](https://github.com/ExchangeUnion/xud/blob/master/lib/orderbook/)
+=============================================================
+
+```
+currencies {
+Id (string - the currency symbol)
+}
+```
+
+This table is automatically maintained by xud with currencies which have a well-synced wallet and open payment channels to minimum one other xud.
+
+```
+pairs {
+Id (string - `{baseCurrencyId}/{quoteCurrencyId}`)
+baseCurrencyId
+quoteCurrencyId
+swapProtocol ({LND, RAIDEN})
+}
+```
+
+This table is automatically maintained by xud with pairs which are swap-enabled via payment channels.
+
+```
+Orders {
+Id
+pairId
+peerId
+quantity (decimal)
+price (decimal)
+]
+```
+
+Table will maintain the open orders which are received by peer nodes (defined in Peers table in the P2P domain database). A peerID is null for the exchange's own orders, the quantity is positive for buy orders, negative for sell orders.
+
+```
+orderThresholds {
+Id
+pairId
+type({disable, peerId, minPrice, maxPrice, minQuantity})
+value
+}
+```
+
+This table defines filters for orders which are preferable to the exchange. For example filter out orders from certain peers or filter out orders below a minimum quantity, which exchanges typically have defined individually for each pair.
+
```

Comment:
Need to emphasis that these filtering thresholds apply in the context of a specific `pairId`.
That explains why `peerId` option is there, while blocking of peers is to be defined in the 'Peers' table (P2P domain).

Reply:
Fixed that.